drivers: timer: Increase default user-allocable nrf_rtc_timer channel count

AdamZelikNS opened the pull request:
Increase the default user-allocable number of RTC channels to 3 when nrf_802154 radio driver is enabled.

rationale: New features in the nrf_802154 will need an additional channel.

AdamZelikNS committed:
drivers: timer: Increase default user-allocable channel count to 3

Increase the default user-allocable number of RTC channels to meet
the nrf_802154 driver requirements.

Signed-off-by: Adam Zelik <adam.zelik@nordicsemi.no>

andyross reviewed (dismissed):
Sure, with the obligatory "hardware details like number of RTCs belong in devicetree somewhere" note.

AdamZelikNS commented:
rebsed, added `samples/boards/nrf/ieee802154/802154_rpmsg/prj.conf` update

AdamZelikNS commented:
@anangl I am changing a few kconfig numbers, please take a look
(also downstream [pr#875](https://github.com/nrfconnect/sdk-zephyr/pull/875))

anangl commented on samples/boards/nrf/ieee802154/802154_rpmsg/prj.conf:
```diff
@@ -5,4 +5,4 @@ CONFIG_UART_CONSOLE=n
 
 CONFIG_NRF_802154_SER_RADIO=y
-CONFIG_NRF_RTC_TIMER_USER_CHAN_COUNT=2
+CONFIG_NRF_RTC_TIMER_USER_CHAN_COUNT=3
```
Why not just remove this entry? `NRF_802154_SER_RADIO` selects `NRF_802154_RADIO_DRIVER`, so `NRF_RTC_TIMER_USER_CHAN_COUNT` will be 3 by default.

AdamZelikNS replied:
Ok, I like such a cleanup. Entry removed.
However, in the case of `CONFIG_NRF53_SYNC_RTC=y`, it turned out that the default value is from `soc/arm/nordic_nrf/nrf53/Kconfig.soc`, so I made a change there.

AdamZelikNS committed:
samples: boards: nrf: ieee802154: Remove RTC user channel count setting

Setting CONFIG_NRF_RTC_TIMER_USER_CHAN_COUNT here is unnecessary
as the value needed is equal to the default.

Signed-off-by: Adam Zelik <adam.zelik@nordicsemi.no>

jciupis commented on soc/arm/nordic_nrf/nrf53/Kconfig.soc:
```diff
@@ -148,6 +148,7 @@ config NRF53_SYNC_RTC_INIT_PRIORITY
 	  nRF53 Synchronized RTC initialization priority.
 
 config NRF_RTC_TIMER_USER_CHAN_COUNT
+	default 3 if NRF_802154_RADIO_DRIVER
```
If we need three channels for nRF 802.15.4 Radio Driver and one more for sync RTC: https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/nordic_nrf/nrf53/sync_rtc.c#L244, then we should have `default 4` in this line, right?

AdamZelikNS replied:
sync_rtc allocates and uses the channel only during initialization, then frees it. The 802154 driver can then allocate this channel when an operation using hw-task is requested. So `default 3` is enough.

jciupis replied:
Fair enough.

jciupis replied:
Perhaps a comment with the above explanation would be nice here to avoid misunderstandings in the future.
